shoes-swt/lib/shoes/swt/arrow_painter.rb

```diff
@@ -0,0 +1,36 @@
+class Shoes
+  module Swt
+    class ArrowPainter < Common::Painter
+      def fill(gc)
+        gc.fill_path(path)
+      end
+
+      def draw(gc)
+        gc.draw_path(path)
+      end
+
+      def path
+        @path ||= begin
+          path = ::Swt::Path.new(::Swt.display)
+
+          path.move_to(@obj.left - @obj.width / 2, @obj.top)
+          path.line_to(@obj.left - @obj.width / 2, @obj.top - @obj.width * 0.20)
+          path.line_to(@obj.left + @obj.width * 0.10, @obj.top - @obj.width * 0.20)
+          path.line_to(@obj.left + @obj.width * 0.10, @obj.top - @obj.width * 0.20 - @obj.width * 0.20)
+          path.line_to(@obj.left + @obj.width * 0.50, @obj.top)
+          path.line_to(@obj.left + @obj.width * 0.10, @obj.top + @obj.width * 0.20 + @obj.width * 0.20)
+          path.line_to(@obj.left + @obj.width * 0.10, @obj.top + @obj.width * 0.20)
+          path.line_to(@obj.left - @obj.width / 2, @obj.top + @obj.width * 0.20)
+          path.line_to(@obj.left - @obj.width / 2, @obj.top - @obj.width * 0.20)
+          path.line_to(@obj.left - @obj.width / 2, @obj.top)
```

Comment:
I think I'd prefer all the floats (0.10, 0.20, 0.5) to be constants.

Not totally sure how to make this smaller... it seems though that `width * 0.x` is rather repeated so putting that into local variables might help as it might also have a meaning like... dunnno `arrow_thickness` or something like that.

Looking at it from that angle one _maybe_ could also extract methods like draw_arrowhead/draw_arrowbody. Sadly, I don't really understand it so I'm more guessing :D